src/rules.jl

```diff
@@ -100,15 +100,15 @@ end
 # unary #
 #-------#
 
-@define_diffrule SpecialFunctions.erf(x)         = :(  (2 / sqrt(π)) * exp(-$x * $x)       )
+@define_diffrule SpecialFunctions.erf(x)         = :(  (2 / sqrt(oftype($x, π))) * exp(-$x * $x)       )
 @define_diffrule SpecialFunctions.erfinv(x)      =
-    :(  (sqrt(π) / 2) * exp(SpecialFunctions.erfinv($x)^2)  )
-@define_diffrule SpecialFunctions.erfc(x)        = :( -(2 / sqrt(π)) * exp(-$x * $x)       )
+    :(  (sqrt(oftype($x, π)) / 2) * exp(SpecialFunctions.erfinv($x)^2)  )
+@define_diffrule SpecialFunctions.erfc(x)        = :( -(2 / sqrt(oftype($x, π))) * exp(-$x * $x)       )
 @define_diffrule SpecialFunctions.erfcinv(x)     =
-    :( -(sqrt(π) / 2) * exp(SpecialFunctions.erfcinv($x)^2)  )
-@define_diffrule SpecialFunctions.erfi(x)        = :(  (2 / sqrt(π)) * exp($x * $x)        )
+    :( -(sqrt(oftype($x, π)) / 2) * exp(SpecialFunctions.erfcinv($x)^2)  )
+@define_diffrule SpecialFunctions.erfi(x)        = :(  (2 / sqrt(oftype($x, π))) * exp($x * $x)        )
 @define_diffrule SpecialFunctions.erfcx(x)       =
-    :(  (2 * $x * SpecialFunctions.erfcx($x)) - (2 / sqrt(π))  )
+    :(  (2 * $x * SpecialFunctions.erfcx($x)) - (2 / sqrt(oftype($x, π)))  )
```

Comment:
I think this could fail for integer inputs (unless there is some promotion before the rules are hit). Generally, it's safer to use "output" variables for the promotion so it would be something like
```
y = 2 * $x * SpecialFunctions.erfcx($x))
return y - oftype(y, 2)/π
```

Reply:
Sorry for the delay here; been a busy couple of months and so this was pushed quite far down my priority list.

But yeah, this is a good point. Is the best solution then to go through all of the rules and manually make these changes, or is there a better way?